web/modules/custom/joinup_notification/src/EventSubscriber/NotificationSubscriberBase.php

```diff
@@ -204,6 +205,49 @@ protected function getUsersMessages(array $user_data, EntityInterface $entity =
     return $return;
   }
 
+  /**
+   * Retrieves a list of emails from a given list of roles.
+   *
+   * @param \Drupal\Core\Entity\EntityInterface $entity
+   *   The group entity.
+   * @param array $bcc_data
+   *   An associative array of roles indexed by the 'roles' key or the
+   *   'og_roles' key depending on whether it is a site wide role or an og role.
+   * @param array $uids_to_skip
+   *   (optional) An array of ids to skip.
+   *
+   * @return array
+   *   An array of emails.
+   */
+  protected function getBccEmails(EntityInterface $entity, array $bcc_data, array $uids_to_skip = []): array {
+    $return = [];
+    foreach ($bcc_data['roles'] as $role_id) {
+      $uids = $this->getRecipientIdsByRole($role_id);
+      if ($uids = array_diff(array_values($uids), $uids_to_skip)) {
+        $emails = array_map(function (UserInterface $user): string {
+          return $user->getEmail();
+        }, $this->entityTypeManager->getStorage('user')->loadMultiple($uids));
```

Comment:
`$this->entityTypeManager->getStorage('user')` is computed on each iteration. Why not setting a var?